### PR TITLE
Ports: Update all port patches to use git format-patch style patches / Meta: Enforce git patches for ports

### PR DESCRIPTION
**Pull request by alimpfard**
This PR removes every. single. non-git patch. in. existence, and bans such patches going forward.
I will not accept another patch with no rhyme or reason, I refuse to do this again.

There are some patches that are split somewhat arbitrarily as the original intent was not clear or just not explained at all, these and some more are marked as FIXME which should ideally be fixed going forward as we continue updating ports.

### CODE REVIEW

**Comment by alimpfard** [comment]
Note: waiting for @BertalanD to split up the GCC patches to complete this massive patch set :)

File: Toolchain/Patches/gcc/0001-Add-a-gcc-driver-for-SerenityOS.patch [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+Co-Authored-By: Gunnar Beutner <gbeutner@serenityos.org>
+Co-Authored-By: Itamar <itamar8910@gmail.com>
+Co-Authored-By: Daniel Bertalan <dani@danielbertalan.dev>
+Co-Authored-By: Nico Weber <thakis@chromium.org>
+Co-Authored-By: Tim Schumacher <timschumi@gmx.de>
+Co-Authored-By: Andrew Kaster <andrewdkaster@gmail.com>
+Co-Authored-By: Brian Gianforcaro <bgianf@serenityos.org>
+Co-Authored-By: Philip Herron <herron.philip@googlemail.com>
+Co-Authored-By: Shannon Booth <shannon.ml.booth@gmail.com>
```
Comment (alimpfard) [inline Toolchain/Patches/gcc/0001-Add-a-gcc-driver-for-SerenityOS.patch:20]: Commit archeology performed by `git log -L31,170:../Toolchain/Patches/gcc.patch --pretty=short | git shortlog -es`.

File: Ports/ruby/version.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 RUBY_MAJOR="3.0"
-RUBY_TEENY="2"
+RUBY_TEENY="4"
```
Comment (timschumi) [inline Ports/ruby/version.sh:2]: Was this included by accident?

Comment (alimpfard) [reply Ports/ruby/version.sh:2]: Not really, I figured updating the port and removing one of the patches is easier than rebuilding ruby, updating the patches, then updating ruby and removing one of the patches.

**Review by danners — COMMENTED** [review]
Typo in commit, it's x264 not x265

**Comment by alimpfard** [comment]
Changelog:
- Fixed the weird thing where the author was also a co-author in some ports (import script boog)
- Reworded the x264 commit to spell "x264" and not "x265"

**Comment by alimpfard** [comment]
Another changelog:
- Omitted "Co-Authored-By" trailers in generated ReadMe files.

**Pull request merged**